File: testdata/SampleSolution/SampleApp/Models/Order.cs
namespace SampleApp.Models;

/// <summary>Represents a customer order.</summary>
public class Order : AuditableEntity
{
    public string CustomerId { get; set; } = string.Empty;
    public OrderStatus Status { get; set; }
    public decimal Total { get; private set; }

    private readonly List<string> _items = [];

    public IReadOnlyList<string> Items => _items;

    public void AddItem(string item, decimal price)
    {
        _items.Add(item);
        Total += price;
    }

    public void Cancel()
    {
        if (Status == OrderStatus.Shipped || Status == OrderStatus.Delivered)
            throw new InvalidOperationException("Cannot cancel a shipped or delivered order.");
        Status = OrderStatus.Cancelled;
    }
}

File: testdata/SampleVbSolution/SampleVbApp/Models/Order.vb
Namespace Models

    ''' <summary>Represents a customer order.</summary>
    Public Class Order
        Implements IEntity

        Public ReadOnly Property Id As Integer Implements IEntity.Id
        Public ReadOnly Property CreatedAt As DateTime Implements IEntity.CreatedAt
        Public Property CustomerId As Integer
        Public Property Status As OrderStatus
        Public Property Total As Money
        Public Property Items As New List(Of String)

        Public Sub New(id As Integer, customerId As Integer)
            Me.Id = id
            Me.CustomerId = customerId
            Me.CreatedAt = DateTime.UtcNow
            Me.Status = OrderStatus.Pending
        End Sub

        Public Sub AddItem(item As String)
            Items.Add(item)
        End Sub
    End Class

End Namespace
